reporting/README.md

```diff
@@ -0,0 +1,46 @@
+## Testing plan ##
+
+### Reporting API ###
+
+https://w3c.github.io/reporting/#generic-reporting
+
+- test that the `content-type` request header is application/reports+json
+- test that the HTTP method is POST
+
+https://w3c.github.io/reporting/#try-delivery
+
+- check the 'Origin' request header
+- spec bug? where does it set the method to POST?
+
+https://w3c.github.io/reporting/#queue-report
+
+- change the value of navigator.userAgent, check that the original value is reported
+- test that report destination URL's username/password, fragment are stripped
+- test a URL that doesn't parse
+- test javascript: mailto: data: ws: wss: etc URLs
+- test settings objects
+
+https://w3c.github.io/reporting/#notify-observers
+
+- test report buffer > 100 reports
+  tests: bufferSize.html
+
+https://w3c.github.io/reporting/#add-report
+
+- test that the right report types are visible or not visible to ReportingObserver
+- test `options` with a `types` member that throws on getting
```

Comment:
I'm under the impression that the `options` object will always generated from specification language, which would make this condition impossible. Take that with a grain of salt, though; I haven't reviewed the relevant algorithms in quite some time.

Reply:
`options` is passed to the `ReportingObserver()` constructor by author JS, so it can be any value. This would be more of a WebIDL test, though. Also, `options` is a dictionary and `types` is a `sequence<DOMString>`, so a `types` getter that is a function doesn't get to throwing the exception because a function isn't iterable... But these two cases should throw (when the constructor is called, because webidl conversion of the arguments):
```
new ReportingObserver(() => {}, { types: { *[Symbol.iterator]() { throw 1; } } });
new ReportingObserver(() => {}, { types: [ { valueOf: () => { throw 2; } } ] });
```

https://heycam.github.io/webidl/#create-sequence-from-iterable

Demo: http://software.hixie.ch/utilities/js/live-dom-viewer/saved/8942

Only the first one throws in chromium and gecko, though. Are they waiting with actually iterating the sequence? Or am I misunderstanding the WebIDL spec?